packages/components/organisms/f-checkout/src/components/_tests/Checkout.test.js

```diff
@@ -3210,6 +3237,142 @@ describe('Checkout', () => {
                 });
             });
         });
+
+        describe('getReferralState ::', () => {
+            afterEach(() => {
+                jest.clearAllMocks();
+            });
+
+            it('should retrieve the referral state from the cookie', () => {
+                // Arrange
+                const getSpy = jest.fn().mockImplementation(() => ({ menuReferralState: 1 }));
+
+                const wrapper = shallowMount(VueCheckout, {
+                    store: createStore({ ...defaultCheckoutState, restaurant: { id: '99999' } }),
+                    i18n,
+                    localVue,
+                    propsData,
+                    mocks: {
+                        $logger,
+                        $cookies: {
+                            get: getSpy
+                        }
+                    }
+                });
+
+                // Act
+                wrapper.vm.getReferralState();
+
+                // Assert
+                expect(getSpy).toHaveBeenCalledWith('je-rw-menu-referral-state-99999');
+            });
+
+            describe('when the cookie exists', () => {
+                describe('AND the `menuReferralState` is `1`', () => {
+                    it('should return "ReferredByWeb"', () => {
+                        // Arrange
+                        const getSpy = jest.fn().mockImplementation(() => ({ menuReferralState: 1 }));
+
+                        const wrapper = shallowMount(VueCheckout, {
+                            store: createStore(),
+                            i18n,
+                            localVue,
+                            propsData,
+                            mocks: {
+                                $logger,
+                                $cookies: {
+                                    get: getSpy
+                                }
+                            }
+                        });
+
+                        // Act
+                        const result = wrapper.vm.getReferralState();
+
+                        // Assert
+                        expect(result).toBe('ReferredByWeb');
+                    });
+                });
+
+                describe('AND the `menuReferralState` is not `1`', () => {
```

Comment:
```suggestion
                describe('and the `menuReferralState` is not `1`', () => {
```
AND looks like an acronym to me. 😬

Reply:
Whereas I kinda agree, several of our tests follow that convention already, so I'm inclined to leave it as it is, personally.